File: services/geoapi/geoapi/Utils/ISettings.vb
Imports System

Namespace Utils
    Public Interface ISettings
        Property ParallelismDegree As Integer
        Property Port As Integer
        Property MongoDBConnectionString As String
        Property AgentsCollectionName As String
        Property MongoDBName As String
        Property TObjectsCollectionName As String
        Property MaxContentLength As Integer
        Property AgentIdSize As Integer
        Property ObjectIdSize As Integer
        Property SearchLimit As Integer
        Property TTL As TimeSpan
    End Interface
End Namespace

File: services/geoapi/geoapi2/Utils/ISettings.cs
using System;

namespace geoapi2.Utils
{
    public interface ISettings
    {
        int ParallelismDegree { get; set; }
        int Port { get; set; }
        string MongoDBConnectionString { get; set; }
        string AgentsCollectionName { get; set; }
        string MongoDBName { get; set; }
        string TObjectsCollectionName { get; set; }

        int MaxContentLength { get; set; }

        int AgentIdSize { get; set; }
        int ObjectIdSize { get; set; }
        int SearchLimit { get; set; }
        TimeSpan TTL { get; set; }
    }
}